File: bit-vectors.js
const bitVector=function(length){
	this.v=new Uint8Array(length>>3);
	this.toggleBit=(pos)=>{this.v[pos>>3]^=1<<pos%8;};
	this.getBit=(pos)=>{return this.v[pos>>3]&1<<pos%8?true:false;};
};


const bv=new bitVector(16);

for(let i=0;i<16;i++){
	bv.toggleBit(i);
	console.log(bv.getBit(i));
	bv.toggleBit(i);
	console.log(bv.getBit(i));
}

File: bit-vectors.go
package bitVector

type BitVectorType struct {
  BitVector []byte
}

func (v *BitVectorType) toggleBit(pos uint) {
  v.BitVector[pos >> 3] ^= 1 << (pos - 1 % 8)
}

func (v *BitVectorType) getBit(pos uint) bool {
  if v.BitVector[pos >> 3] & (1 << (pos - 1 % 8)) == 0 {
    return false
  }
  return true
}

func (b *BitVectorType) Init(length int) *BitVectorType {
  b.BitVector = make([]byte, length >> 3)
  return b
}

func New(length int) *BitVectorType {
  return new(BitVectorType).Init(length)
}
